go-controller/pkg/clustermanager/userdefinednetwork/template/net-attach-def-template_test.go

```diff
@@ -273,10 +273,12 @@ var _ = Describe("NetAttachDefTemplate", func() {
 			udnv1.UserDefinedNetworkSpec{
 				Topology: udnv1.NetworkTopologyLayer2,
 				Layer2: &udnv1.Layer2Config{
-					Role:          udnv1.NetworkRolePrimary,
-					Subnets:       udnv1.DualStackCIDRs{"192.168.100.0/24", "2001:dbb::/64"},
-					MTU:           1500,
-					IPAMLifecycle: udnv1.IPAMLifecyclePersistent,
+					Role:    udnv1.NetworkRolePrimary,
+					Subnets: udnv1.DualStackCIDRs{"192.168.100.0/24", "2001:dbb::/64"},
+					MTU:     1500,
+					IPAM: &udnv1.IPAMSpec{
```

Comment:
Shouldn't spec specify both `IPAM.Mode=Enabled` & `Lifecycle=Persistent`?
Isn't it the intention of the API change?

Should we have additional negative tests, for covering case when IPAM.Mode=Disable & Lifecycle=Persistent?

Reply:
Enables is the default value, so you don't have to set it, but if you do put Disabled there, Persistent lifecycle won't be allowed anymore. I have added a validation test to NAD-template

Reply:
I thought the idea was to have explicit IPAM mode whether it enable or disabled.
But it seem one would not have to set mode=enabled and just define subnet, similar to what we have today.
Did I got it right?

Reply:
yes, and another "new" effect is that you can't set an empty subnet anymore without explicitly setting disabled IPAM mode